xls/passes/select_simplification_pass.cc

```diff
@@ -964,6 +966,41 @@ absl::StatusOr<bool> SimplifyNode(Node* node, const QueryEngine& query_engine,
     }
   }
 
+  // Specialize the arms of the select based on the selector value which chooses
+  // the arm.
+  if (node->Is<Select>()) {
+    Select* sel = node->As<Select>();
+    auto selr_width = sel->selector()->BitCountOrDie();
```

Comment:
Use auto when the type is redundant (like when assigning from make_unique) but generally use the full type to improve readability. Ie, "int64 selr_width =..".